### PR TITLE
3.0 plugin testing

### PR DESCRIPTION
**Pull request by dereuromark**
Stab at adding few missing infos
- https://github.com/cakephp/cakephp/issues/12727
- routing change - migration guide update necessary

Please help to enrich the part about plugin testing.
I feel like - also from the feedback of others in the chat - that we have some infos missing here.

### CODE REVIEW

File: en/plugins.rst [hunk 1]
```diff
@@ -747,7 +747,17 @@ You can customize the command names by defining each command in your plugin::
...
 
 
-Publishing Your Plugin
+Testing your Plugin
+===================
+
+...
+
+If you have to test controllers or URL generation in general, make sure you
+wire up some routing in your `tests/bootstrap.php`.
```
Comment (markstory) [inline en/plugins.rst]:
```suggestion
connect some routes in your ``tests/bootstrap.php``.
```

File: en/plugins.rst [hunk 2]
```diff
@@ -747,7 +747,17 @@ You can customize the command names by defining each command in your plugin::
...
+
+For more information see :ref:`testing plugins <testing.rst> page.
```
Comment (markstory) [inline en/plugins.rst]:
```suggestion
For more information see :ref:`testing plugins </development/testing.rst> page.
```

**Comment by dereuromark** [comment]
We could add the info about

    vendor/bin/phpunit plugins/tests/TestCase/

call from app root
this should suffice as main entry point for the command.

Otherwise, any other useful test harness setup infos we could add?

File: en/plugins.rst [hunk 3]
```diff
@@ -747,7 +747,17 @@ You can customize the command names by defining each command in your plugin::
     }
 
 
-Publishing Your Plugin
+Testing your Plugin
+===================
+
+...
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply en/plugins.rst]: Placeholder to add more useul infos here.
Anyone ideas?

[the remark this answers is not in the document]
Comment (markstory) [reply en/plugins.rst]: The empty placeholder text isn't going to be helpful to the reader. We could remove it for now.

**Pull request merged**